face/FindFaces/python/find_faces_in_collection.ipynb

```diff
@@ -0,0 +1,162 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "id": "759f9ec0",
+   "metadata": {},
+   "source": [
+    "# API Face Samples"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "61534b8b",
+   "metadata": {},
+   "source": [
+    "## Objective\n",
+    "\n",
+    "To develop a Python-based application that detects and verifies faces within a face collection (face list or person group) using Azure Face SDK and REST API. The application will cover three scenarios:\n",
+    "\n",
+    "- **Image + Face List**: Detect faces in the image and check if the largest detected face matches any face in the given face list.\n",
+    "- **Face ID + Face List**: Verify if the provided face ID matches any face in the given face list.\n",
+    "- **Image + Person Group**: Detect faces in the image and identify if the largest detected face belongs to any person in the given person group.\n",
+    "- **Face ID + Person Group**: Identify if the provided face ID belongs to any person in the given person group.\n",
+    "\n",
+    "The function returns relevant identifiers and confidence scores, allowing users to determine the match accuracy and identity of faces in different contexts.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "821406a6",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os, requests\n",
+    "from azure.core.credentials import AzureKeyCredential\n",
+    "from azure.ai.vision.face import FaceClient\n",
+    "from azure.ai.vision.face.models import FaceDetectionModel, FaceRecognitionModel\n",
+    "\n",
+    "def find_faces_in_collection(subscription_key, endpoint, image_path=None, face_id=None, face_list_id=None, person_group_id=None):\n",
+    "    headers = {\n",
+    "        'Ocp-Apim-Subscription-Key': subscription_key,\n",
+    "        'x-ms-useragent': 'find-faces-in-collection/1.0.0'\n",
+    "    }\n",
+    "    \n",
+    "    def detect_face(image_path):\n",
+    "        with FaceClient(endpoint, AzureKeyCredential(subscription_key), headers = {\"x-ms-useragent\": \"find-faces-in-collection/1.0.0\"} ) as face_client:\n",
+    "            with open(image_path, 'rb') as image_data:\n",
+    "                detected_faces = face_client.detect(\n",
+    "                        image_content=image_data,\n",
+    "                        detection_model=FaceDetectionModel.DETECTION_03,\n",
+    "                        recognition_model=FaceRecognitionModel.RECOGNITION_04,\n",
+    "                        return_face_id=True,\n",
+    "                    )\n",
+    "            return [face['faceId'] for face in detected_faces]\n",
```

Comment:
While the returned `FaceDetectionResult` object does allow syntax sugar `__getitem__` to access/lookup underlying JSON response, SDK owners suggest to retrieve the structed info via predefined instance variable.

```suggestion
    "                detected_faces = face_client.detect(\n",
    "                        image_content=image_data.read(),\n",
    "                        detection_model=FaceDetectionModel.DETECTION_03,\n",
    "                        recognition_model=FaceRecognitionModel.RECOGNITION_04,\n",
    "                        return_face_id=True,\n",
    "                    )\n",
    "            return [face.face_id for face in detected_faces]\n",
```